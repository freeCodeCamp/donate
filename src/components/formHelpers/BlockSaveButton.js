import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

function BlockSaveButton(props) {
  return (
    <Button block={true} bsStyle='primary' {...props} type='submit'>
      {props.children || 'Save'}
    </Button>
  );
}

BlockSaveButton.displayName = 'BlockSaveButton';
BlockSaveButton.propTypes = {
  children: PropTypes.any
};

export default BlockSaveButton;
