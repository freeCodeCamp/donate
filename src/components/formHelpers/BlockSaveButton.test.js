/* global expect */

import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BlockSaveButton from './BlockSaveButton';

Enzyme.configure({ adapter: new Adapter() });

test('<BlockSaveButton /> snapshot', () => {
  const component = renderer.create(<BlockSaveButton />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button text should default to "Save"', () => {
  const enzymeWrapper = Enzyme.render(<BlockSaveButton />);

  expect(enzymeWrapper.text()).toBe('Save');
});

test('Button text should match "children"', () => {
  const enzymeWrapper = Enzyme.render(
    <BlockSaveButton>My Text Here</BlockSaveButton>
  );

  expect(enzymeWrapper.text()).toBe('My Text Here');
});
