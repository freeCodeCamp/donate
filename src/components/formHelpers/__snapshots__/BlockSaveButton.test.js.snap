// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockSaveButton /> snapshot 1`] = `
<button
  className="btn btn-primary btn-block"
  disabled={false}
  type="submit"
>
  Save
</button>
`;
